oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/MissingLastRevSeekerTest.java

```diff
@@ -216,4 +220,26 @@ public void getClusterNodeInfo() {
     private ClusterNodeInfoDocument getClusterNodeInfo(int clusterId) {
         return seeker.getClusterNodeInfo(clusterId);
     }
+    
+    @Test
+    public void getNonSplitDocs() throws Exception {
+        String nodeName = this.getClass().getName() + "-foo";
+        DocumentNodeStore dns = getBuilder().setDocumentStore(store).getNodeStore();
+        NodeBuilder b1 = dns.getRoot().builder();
+        b1.child(nodeName);
+        dns.merge(b1, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+        //Modify and commit changes on this node 100 times to create a split document
+        for (int i = 0; i < NUM_REVS_THRESHOLD; i++) {
+            b1 = dns.getRoot().builder();
+            b1.child(nodeName).setProperty("prop",i);
+            dns.merge(b1, EmptyHook.INSTANCE, CommitInfo.EMPTY);
+        }
+        dns.runBackgroundOperations();
+        int docs = 0;
+        //seeker should return only non split documents
+        for(Document doc : seeker.getCandidates(0)) {
+        	docs++;
+        }
+        assertEquals(2, docs);
+    }
```

Comment:
The DocumentNodeStore is not disposed.
```suggestion
        dns.dispose();
    }
```

Reply:
As per above review comment we should not do this right?

Reply:
Actually, the test _should_ dispose the `DocumentNodeStore` when it is not needed anymore. Some tests use a `DocumentMKBuilderProvider` rule that takes care of disposing the node store automatically after the test, but this test creates the node store manually.

Reply:
Ok so we do need to dispose the DocumentNodeStore but not underlying DocumentStore for this test case. I was confused with these two comments.